src/ff8/vram.cpp

```diff
@@ -291,36 +301,141 @@ int read_vram_to_buffer_with_palette2_parent_call(int a1, uint16_t rel_pos, int1
+void set_tex_name(const TexturePacker::TiledTex &tiledTex, ff8_tex_header *tex_header)
+{
+	if (tex_header == nullptr)
+	{
+		return;
+	}
+
+	const auto textures = texturePacker.matchTextures(tiledTex);
+
+	if (textures.empty())
+	{
+		return;
+	}
+
+	std::string filename;
+	int file_count = 0;
+	int startVramId = 0;
+
+	for (const TexturePacker::IdentifiedTexture &tex: textures)
+	{
+		if (tex.isValid())
+		{
+			if (file_count == 0)
+			{
+				filename.append(tex.name());
+				startVramId = tex.texture().vramId();
+			}
+			else
+			{
+				size_t index = tex.name().find_last_of('/');
+				if (index < 0)
+				{
+					filename.append(tex.name());
+				}
+				else
+				{
+					filename.append(tex.name().substr(index + 1));
+				}
+			}
+			filename.append("_");
+			file_count += 1;
+
+			// Do not produce huge file names
+			if (file_count > 6)
+			{
+				break;
+			}
+		}
+	}
+
+	if (!filename.empty())
+	{
+		int vramId = next_psxvram_x / TEXTURE_WIDTH_BPP16 + (next_psxvram_y >= TEXTURE_HEIGHT ? 16 : 0);
+
+		if (file_count == 1) {
+			// To be on par with remaster naming, instead for example with use "card_16", "card_17" etc.., we starts with "card_0"
+			vramId -= startVramId;
+		}
+
+		if (filename.size() > 240) {
+			filename = filename.substr(0, 240);
+
+			if (!filename.ends_with('_')) {
+				filename.append("_");
+			}
+		}
+
+		filename.append(std::to_string(vramId));
+
+		if (trace_all || trace_vram) ffnx_trace("%s: %s\n", __func__, filename.c_str());
+
+		tex_header->file.pc_name = (char*)external_malloc(1024);
+
+		if (tex_header->file.pc_name != nullptr)
+		{
+			strncpy(tex_header->file.pc_name, filename.c_str(), 1024);
+		}
+	}
+}
+
+void free_tex_header(ff8_tex_header *tex_header)
 {
-	if (trace_all || trace_vram) ffnx_trace("%s: vram_pos=(%d, %d) target=0x%X target_w=%d w=%d h=%d bpp=%d\n", __func__, next_psxvram_x, next_psxvram_y, int(target), target_w, w, h, bpp);
+	if (tex_header != nullptr && uint32_t(tex_header->file.pc_name) > 32)
+	{
+		if (trace_all || trace_vram) ffnx_trace("%s: %s\n", __func__, tex_header->file.pc_name);
+
+		external_free(tex_header->file.pc_name);
+	}
+
+	((void(*)(ff8_tex_header*))ff8_externals.psxvram_texture_page_tex_header_free)(tex_header);
+}
+
+void read_vram_to_buffer(uint8_t *vram, int vram_w_2048, uint8_t *target, int target_w_in_bytes, signed int w, int h, int bpp)
+{
+	if (trace_all || trace_vram) ffnx_trace("%s: vram_pos=(%d, %d) target=0x%X target_w_in_bytes=%d w=%d h=%d bpp=%d\n", __func__, next_psxvram_x, next_psxvram_y, int(target), target_w_in_bytes, w, h, bpp);
 
 	if (next_psxvram_x == -1)
 	{
 		ffnx_warning("%s: cannot detect VRAM position\n", __func__);
 	}
 	else
 	{
-		texturePacker.registerTiledTex(target, next_psxvram_x, next_psxvram_y, target_w, h, Tim::Bpp(bpp));
+		const TexturePacker::TiledTex &tiledTex = texturePacker.registerTiledTex(target, next_psxvram_x, next_psxvram_y, w, h, Tim::Bpp(bpp));
```

Comment:
Use "w" instead "target_w" to fix a find texture problem

Reply:
Nope I left this comment for you, so you see one change hidden in the diff of another change

Reply:
Ah all right, usually Github resolves this automatically when it detects a change but it didn't for this one :) Thanks

Reply:
Not if I make the comment after the change :)